modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account/domains.jsp

```diff
@@ -0,0 +1,185 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<%
+AccountDisplay accountDisplay = (AccountDisplay)request.getAttribute(AccountWebKeys.ACCOUNT_DISPLAY);
+
+List<String> domains = Collections.emptyList();
+
+if (accountDisplay != null) {
+	domains = accountDisplay.getDomains();
+}
+%>
+
+<liferay-util:buffer
+	var="removeDomainIcon"
+>
+	<liferay-ui:icon
+		icon="times-circle"
+		markupView="lexicon"
+		message="remove"
+	/>
+</liferay-util:buffer>
+
+<div class="sheet-section">
+	<h3 class="autofit-row sheet-subtitle">
+		<span class="autofit-col autofit-col-expand">
+			<span class="heading-text"><liferay-ui:message key="valid-domains" /></span>
+		</span>
+		<span class="autofit-col">
+			<span class="heading-end">
+				<liferay-ui:icon
+					cssClass="modify-link"
+					id="addDomains"
+					label="<%= true %>"
+					linkCssClass="btn btn-secondary btn-sm"
+					message="add"
+					method="get"
+					url="javascript:;"
+				/>
+			</span>
+		</span>
+	</h3>
+
+	<aui:input name="domainNames" type="hidden" value="<%= StringUtil.merge(domains) %>" />
+
+	<liferay-ui:search-container
+		compactEmptyResultsMessage="<%= true %>"
+		emptyResultsMessage="this-account-does-not-have-any-valid-domain"
+		headerNames="title,null"
+		id="accountDomainsSearchContainer"
+		iteratorURL="<%= currentURLObj %>"
+		total="<%= domains.size() %>"
+	>
+		<liferay-ui:search-container-results
+			results="<%= domains.subList(searchContainer.getStart(), searchContainer.getResultEnd()) %>"
+		/>
+
+		<liferay-ui:search-container-row
+			className="java.lang.String"
+			modelVar="domain"
+		>
+			<liferay-ui:search-container-column-text
+				cssClass="table-cell-content"
+				name="name"
+				value="<%= domain %>"
+			/>
+
+			<liferay-ui:search-container-column-text>
+				<a class="modify-link pull-right" data-rowId="<%= domain %>" href="javascript:;"><%= removeDomainIcon %></a>
+			</liferay-ui:search-container-column-text>
+		</liferay-ui:search-container-row>
+
+		<liferay-ui:search-iterator
+			markupView="lexicon"
+		/>
+	</liferay-ui:search-container>
+</div>
+
+<aui:script use="liferay-search-container">
+	var searchContainer = Liferay.SearchContainer.get('<portlet:namespace />accountDomainsSearchContainer');
+
+	var searchContainerContentBox = searchContainer.get('contentBox');
+
+	var domainNamesInput = document.<portlet:namespace />fm.<portlet:namespace />domainNames;
+
+	var domainNames = domainNamesInput.value.split(',');
```

Comment:
Update to `var domainNames = domainNamesInput.value.split(',').filter(Boolean);` to avoid a blank string in the array